From 4e548d7e9a5dcae911c1590e3e0c69cc6bee427d Mon Sep 17 00:00:00 2001
From: jhb <jhb@FreeBSD.org>
Date: Sun, 7 Oct 2012 18:42:02 +0000
Subject: [PATCH 8/9] MFC 240962: Grab the mfi_config_lock while performing a
 MFI_DCMD_CFG_FOREIGN_IMPORT request on behalf of a user
 utility.

---
 sys/dev/mfi/mfi.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/sys/dev/mfi/mfi.c b/sys/dev/mfi/mfi.c
index f801cde..c1ddb5c 100644
--- a/sys/dev/mfi/mfi.c
+++ b/sys/dev/mfi/mfi.c
@@ -2550,6 +2550,7 @@ mfi_config_lock(struct mfi_softc *sc, uint32_t opcode)
 	case MFI_DCMD_LD_DELETE:
 	case MFI_DCMD_CFG_ADD:
 	case MFI_DCMD_CFG_CLEAR:
+	case MFI_DCMD_CFG_FOREIGN_IMPORT:
 		sx_xlock(&sc->mfi_config_lock);
 		return (1);
 	default:
-- 
1.7.10.4

